monai/data/wsi_reader.py

```diff
@@ -827,6 +911,36 @@ def _get_patch(
             patch = np.take(patch, [0, 1, 2], self.channel_dim)
 
         return patch
+    
+    def _resize_to_mpp_res(self, wsi, closest_lvl, mpp_list, user_mpp: tuple):
```

Comment:
This is definitely better to reduce duplication, however this and the other method of the same name below are virtually identical except for some type conversions. I think this method could be moved to `BaseWSIReader` and then here we'd have an override method calling the base one which does the conversion of outputs:

```python
def _resize_to_mpp_res(self, wsi, closest_lvl, mpp_list, user_mpp: tuple):
  cucim_resize, _ = optional_import("cucim.skimage.transform", name="resize")
  cp, _ = optional_import("cupy")

  closest_lvl_wsi, target_res_x, target_res_y = super()._resize_to_mpp_res(wsi, closest_lvl, mpp_list, user_mpp)
  wsi_arr = cp.array(closest_lvl_wsi)
  closest_lvl_wsi = cucim_resize(wsi_arr, (target_res_x, target_res_y), order=1)

  return closest_lvl_wsi
```

The `_resize_to_mpp_res` version in `BaseWSIReader` wouldn't have any Numpy/Cupy code in it but just implement the base algorithm and return the result plus the target resolution values. The version in `OpenSlideWSIReader` would use PIL to do the resizing instead of Cupy. 

Does this make sense to do?

Reply:
That sounds like a good solution to me. However, I would keep the computation of `closest_lvl_dim` and `closest_lvl_wsi` in the class specific function, and use `super()._resize_to_mpp_res` to compute `target_res_x` and `target_res_y`.

Reply:
Sure, feel free to update things.

Reply:
Hi Eric, I have added a function to compute the target resolution to the BaseWSIReader class. Based on your idea, _resize_to_mpp_res then calls super()._compute_target_res, which returns target_x and target_y. I would suggest to keep the image extraction methods (read_region etc.) in _resize_to_mpp_res since they depend on the backend. Feel free to check out the result. Though, I still have to modify the docstring of _compute_target_res.

Reply:
This looks much better with much less redundancy, thanks!